plugins/aws/resoto_plugin_aws/resource/cloudfront.py

```diff
@@ -0,0 +1,1429 @@
+            builder.add_edge(
+                self, clazz=AwsIamServerCertificate, id=self.distribution_viewer_certificate.iam_certificate_id
+            )
+
+        # TODO edge to ACM certificate when applicable (via self.distribution_viewer_certificate.acm_certificate_arn)
+        # TODO edge to Web Acl when applicable (via self.distribution_web_acl_id)
+
+    # TODO implement this after more research
+    # def delete_resource(self, client: AwsClient) -> bool:
+    #     client.call(aws_service=self.api_spec.service, action="delete-distribution", result_name=None, Id=self.id)
+    #     return True
+
+
+@define(eq=False, slots=False)
+class AwsCloudFrontFunctionConfig:
+    kind: ClassVar[str] = "aws_cloudfront_function_config"
+    mapping: ClassVar[Dict[str, Bender]] = {"comment": S("Comment"), "runtime": S("Runtime")}
+    comment: Optional[str] = field(default=None)
+    runtime: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsCloudFrontFunction(CloudFrontTaggable, CloudFrontResource):
+    kind: ClassVar[str] = "aws_cloudfront_function"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("cloudfront", "list-functions", "FunctionList")
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("Name"),
+        "arn": S("arn"),
+        "name": S("Name"),
+        "ctime": S("ctime"),
+        "mtime": S("mtime"),
+        "function_status": S("Status"),
+        "function_stage": S("stage"),
+        "function_config": S("FunctionConfig") >> Bend(AwsCloudFrontFunctionConfig.mapping),
+    }
+    function_status: Optional[str] = field(default=None)
+    function_config: Optional[AwsCloudFrontFunctionConfig] = field(default=None)
+
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def add_tags(func: AwsCloudFrontFunction) -> None:
+            tags = builder.client.get("cloudfront", "list-tags-for-resource", "Tags", Resource=func.arn)
+            if tags:
+                func.tags = bend(ToDict(), tags["Items"])
+
+        for js in json:
+            js["arn"] = js["FunctionMetadata"]["FunctionARN"]
+            js["stage"] = js["FunctionMetadata"]["Stage"]
+            js["ctime"] = js["FunctionMetadata"]["CreatedTime"]
+            js["mtime"] = js["FunctionMetadata"]["LastModifiedTime"]
+            instance = cls.from_api(js)
+            builder.submit_work(add_tags, instance)
+            builder.add_node(instance, js)
+
+    # TODO implement this after more research
+    # def delete_resource(self, client: AwsClient) -> bool:
+    #     client.call(
+    #       aws_service=self.api_spec.service,
+    #       action="delete-function",
+    #       result_name=None, Name=self.name,
+    #       IfMatch="ETag value")
+    #     return True
+
+
+@define(eq=False, slots=False)
+class AwsCloudFrontInvalidation(CloudFrontResource):
```

Comment:
Those aren't really resources. An invalidation is when you send a purge request to a CF distribution. It's like an http request.

Reply:
@lloesche I included them because they can incur costs: [read here](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.html#PayingForInvalidation)
felt relevant to me 🤷🏻‍♀️